pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ConfigurationDataUtils.java

```diff
@@ -51,20 +51,17 @@ public static ObjectMapper getThreadLocal() {
         return mapper.get();
     }
 
-    private ConfigurationDataUtils() {}
+    private ConfigurationDataUtils() {
+    }
 
     public static <T> T loadData(Map<String, Object> config,
-                                 T existingData,
-                                 Class<T> dataCls) {
+                                 T existingData) {
```

Comment:
Unless I'm mistaken, I believe this is part of the public API and could be used by user applications to build the `ClientConfigurationData`. We would need to find a way to introduce this without breaking the API.

Reply:
+1. We can keep the old method and just mark deprecated.

Reply:
yes, I think it could have been used by users.
we should keep the old, make it work the same way as before and use the new method